dsl/src/parser/AST/AstVisitor.java

```diff
@@ -145,6 +145,16 @@ default T visit(AggregateValueDefinitionNode node) {
         return null;
     }
 
+    /**
+     * Visitor method for FuncDefNodes
+     *
+     * @param node Node to visit
+     * @return T
+     */
+    default T visit(FuncDefNode node) {
+        return null;
```

Comment:
Was ist die Bedeutung von `null` an dieser Stelle? 

Wenn Du den Usern anbieten möchtest, das Interface direkt zu verwenden, ohne alle Methoden implementieren zu müssen (meine Vermutung wg. der Default-Methoden), würde ich in allen Methoden eine `throw new UnsupportedOperationException();` werfen.

Reply:
Alle `AstVisitor`-Implementierungen iterieren und besuchen (im Sinne von `visit()` im Visitor-Pattern) irgendwann mal alle Kind-Knoten eines `Node`. Die Idee ist, dass das auch immer gehen soll und eben nicht alle `visit`-Überladungen implementiert werden müssen, was allerdings nicht funktioniert, wenn standardmäßig eine Exception geworfen wird. 
`null` bedeutet in dem Kontext "ich tu nichts, ich geb nichts zurück und ich bin vorallem nicht im Weg, sodass weiter iteriert werden kann".

Reply:
@cagix Wäre das mit der Erklärung für dich 'resolved'?

Reply:
Ehrlich gesagt verstehe ich die Erklärung grad nicht. Zum einen steuerst Du ja in Deinen konkreten Visitoren, was Du alles besuchst und würdest darüber ja die nicht implementierten Methoden vermeiden. Und andererseits scheinst Du die Rückgabewerte nicht zu nutzen, sonst wäre `null` fatal, oder? Aber es war eh nur eine Anmerkung und nicht eine Bitte um Überarbeitung.

Reply:
@malt-r Ehrlich gesagt verstehe ich die Erklärung grad nicht. Zum einen steuerst Du ja in Deinen konkreten Visitoren, was Du alles besuchst und würdest darüber ja die nicht implementierten Methoden vermeiden. Und andererseits scheinst Du die Rückgabewerte nicht zu nutzen, sonst wäre `null` fatal, oder? Aber es war eh nur eine Anmerkung und nicht eine Bitte um Überarbeitung.